blackbird_python/blackbird/listener.py

```diff
@@ -273,18 +273,21 @@ def exitExpressionvar(self, ctx: blackbirdParser.ExpressionvarContext):
         elif ctx.nonnumeric():
             value = _literal(ctx.nonnumeric())
 
-        try:
-            # assume all variables are scalar
-            final_value = PYTHON_TYPES[vartype](value)
-        except TypeError:
+        if isinstance(value, sym.Expr):
+            final_value = value
```

Comment:
I can't quite tell, but what is the reason for this additional if statement?

Reply:
If the expression turns out to be a parameter, it should just be saved as it is in the variables list (`_VAR`); if not, it will assert type and cast to the proper type if possible.

Reply:
```suggestion
        if isinstance(value, sym.Expr):
            # If the expression is a parameter, save it in the variables list _VAR
            final_value = value
```